conftest.py

```diff
@@ -68,3 +71,22 @@ def component_cache_instance(request):
     # Test was not parametrized, so component instance is not needed
     except AttributeError:
         yield None
+
+
+@pytest.fixture
+def teardown_test_catalog():
+    """
+    This fixture is run after certain tests that modify the component catalog.
+    This way, the catalog instance will be removed even when the test fails
+    """
+    metadata_manager = MetadataManager(schemaspace=ComponentCatalogs.COMPONENT_CATALOGS_SCHEMASPACE_ID)
+
+    # Run test
+    yield
+
+    # Remove test catalog
+    try:
+        if metadata_manager.get(TEST_CATALOG_NAME):
+            metadata_manager.remove(TEST_CATALOG_NAME)
+    except Exception:
+        pass
```

Comment:
This seems a little unusual in that it leads to this mysterious/missing declaration of `metadata_manager` in the tests.  Instead, I think naming this fixture something like `metadata_manager_with_teardown` and adding `metadata_manager` to the `yield` statement would then help clarify things.

The tests would then look something like:
```python
def test_modify_component_catalogs(metadata_manager_with_teardown):
   # do some stuff, build the metadata instance
   metadata_manager_with_teardown.create(TEST_CATALOG_NAME, registry_instance)
   # finish the test
```
Then at least there's _less_ mystery as to where the metadata manager instance came from.

Reply:
That makes sense! I'll re-work those